src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -275,53 +286,111 @@ protected override void ProcessRecord()
         protected override void StopProcessing()
         {
             _sender?.SendAsyncCancel();
+            cancellationTokenSource.Cancel();
         }
 
         #region ConnectionTest
 
+        private void SetCountForTcpTest()
+        {
+            if (Repeat.IsPresent)
+            {
+                Count = int.MaxValue;
+            }
+            else if (!MyInvocation.BoundParameters.ContainsKey("Count"))
+            {
+                Count = 1;
+            }
+        }
+
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
         {
+            SetCountForTcpTest();
+
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
-
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
+                long latency = 0;
+                SocketError status = SocketError.SocketError;
+
+                Stopwatch stopwatch = new Stopwatch();
 
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                using TcpClient client = new TcpClient();
```

Comment:
Why do we create it in every iteration? I think it should be moved out of the cycle to line 314.

Reply:
I played around with this before, I couldn't find a reliable way to get this working 😅

In this first case, we're calling `Close()` after each connection test. Here's the exception thrown during the second loop:
```
Test-Connection: Object reference not set to an instance of an object.
```
Alternatively, if the connection is not closed we run into the following:
```
Test-Connection: Socket is already connected
```
Any idea how we may be able to get around this?

Reply:
Calling `Close()` on an instance of a TcpClient will dispose the instance, which will explain the first error you've got there.

Reply:
Yeah, I don't think these can be re-used; once a connection is closed, the client usually needs to be recreated here.

Reply:
Yeah, thats what I was seeing! Couldn't find any way to close the connection, without also disposing 😔